src/index.ts

```diff
@@ -19,39 +26,88 @@ interface Trie {
   put(key: Buffer | string, value: Buffer | string, cb: TriePutCb): void
 }
 
+interface AccountProps {
+  nonce: Buffer
+  balance: Buffer
+  stateRoot: Buffer
+  codeHash: Buffer
+}
+
+// Would prefer for Account class getters to have type `Buffer`
+// and setters to have type `any` to be cast `toBuffer`,
+// however get ts error:
+// `'get' and 'set' accessor must have the same type`
+type BufferLike = Buffer | any
```

Comment:
When migrating `-tx` to TS we decided to just set all properties as `Buffer`, but keep the `toBuffer` logic in the setters. The rationale was that this is only a breaking change for TS users, who would appreciate the stricter types. Note that we released it as a major version though.

Also, we have a stricter `BufferLike` here: https://github.com/ethereumjs/ethereumjs-tx/blob/master/src/types.ts#L19